src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,45 @@
+package duke.command;
+
+import java.util.ArrayList;
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
+
+public class FindCommand extends Command {
+    private String filter;
+
+    /**
+     * Creates a new FindCommand.
+     *
+     * @param filter    the string to search for
+     */
+    public FindCommand(String filter) {
+        this.filter = filter;
+    }
+
+    public boolean terminate() {
+        return false;
+    }
+
+    public void execute(Storage storage, Ui ui, TaskList taskList) {
+        ArrayList<Integer> matchIndices = new ArrayList<>();
+
+        for (int index = 1; index <= taskList.getSize(); index++) {
+            if (taskList.taskDescriptionContains(index, filter)) {
+                matchIndices.add(index);
+            }
+        }
+
+        ui.printHorizontalLine();
```

Comment:
Would it be possible to print the borders once in run() in Duke class so you do not have to call printHorizontalLine() for every execute()?

Reply:
I suppose it would be possible if I use a `finally` statement to print the closing line.